### PR TITLE
Update bam-to-sam and sam-to-bam to remove need for wrapper script.

### PR DESCRIPTION
**Pull request by davebx**

### CODE REVIEW

**Comment by bgruening** [comment]
@davebx just wondering if an `<stdio>` tag would be useful?
Otherwise this looks good to me and it is a huge improvement!

File: tool_collections/samtools/sam_to_bam/sam_to_bam.xml [hunk 1]
```diff
@@ -1,98 +1,86 @@
-<tool id="sam_to_bam" name="SAM-to-BAM" version="1.1.5">
-  <description>converts SAM format to BAM format</description>
-  <requirements>
-    <requirement type="package" version="1.1">samtools</requirement>
-  </requirements>
-  <command interpreter="python">
-    sam_to_bam.py
-      --input1=$source.input1
+<tool id="bam_to_sam" name="BAM-to-SAM" version="1.0.5">
```
Comment (nsoranzo) [inline tool_collections/samtools/sam_to_bam/sam_to_bam.xml]: Here the version number is decreasing, not good!

File: tool_collections/samtools/sam_to_bam/sam_to_bam.xml [hunk 2]
```diff
@@ -1,98 +1,86 @@
...
+]]>
+    </command>
+    <inputs>
+        <conditional name="source">
+            <param label="Choose the source for the reference list" name="index_source" type="select">
+                <option value="cached">Locally cached</option>
+                <option value="history">History</option>
+            </param>
+            <when value="cached">
+                <param format="sam" label="SAM file to convert" metadata_name="dbkey" name="input" type="data">
```
Comment (nsoranzo) [inline tool_collections/samtools/sam_to_bam/sam_to_bam.xml]: Is it really necessary to rename ```input1``` to ```input```? This breaks tool rerun.

File: tool_collections/samtools/bam_to_sam/bam_to_sam.xml [hunk 3]
```diff
@@ -1,63 +1,49 @@
...
+        <requirement type="package" version="1.1">samtools</requirement>
+    </requirements>
+    <command>
+<![CDATA[
+  samtools sort -O bam -@ \${GALAXY_SLOTS:-1} "$input" -o sorted_input.bam -T temp && 
+  samtools view sorted_input.bam $header -o $output -@ \${GALAXY_SLOTS:-1}
+]]>
+    </command>
+    <inputs>
+        <param format="bam" label="BAM File to Convert" name="input" type="data" />
```
Comment (nsoranzo) [inline tool_collections/samtools/bam_to_sam/bam_to_sam.xml]: Is it really necessary to rename ```input1``` to ```input```? This breaks tool rerun.

File: tool_collections/samtools/sam_to_bam/sam_to_bam.xml (hunk 1 above)
Comment (martenson) [inline tool_collections/samtools/sam_to_bam/sam_to_bam.xml]: also the tool got renamed and re-id'd; this looks like some mismatch

**Comment by davebx** [comment]
@bgruening agreed. Done.

File: tool_collections/samtools/sam_to_bam/sam_to_bam.xml [hunk 4]
```diff
@@ -1,98 +1,89 @@
...
+    <description>converts SAM format to BAM format</description>
+    <requirements>
+        <requirement type="package" version="1.1">samtools</requirement>
+    </requirements>
+    <stdio>
+        <exit_code range="1:" level="fatal" description="Error" />
+    </stdio>
+    <command>
+<![CDATA[
+    samtools sort -O bam -@ \${GALAXY_SLOTS:-1} "$input1" -o sorted_input.bam -T temp &&
```
Comment (nsoranzo) [inline tool_collections/samtools/sam_to_bam/sam_to_bam.xml]: I know this works anyway, but I suggest to `s/$input1/$source.input1/`

Comment (jmchilton) [inline tool_collections/samtools/sam_to_bam/sam_to_bam.xml]: +1

**Comment by martenson** [comment]
closed in favor of #84

**Pull request closed without merging**